src/core/thread/address_resolver.cpp

```diff
@@ -426,12 +449,16 @@ void AddressResolver::HandleAddressError(Coap::Header &aHeader, Message &aMessag
     Mac::ExtAddress macAddr;
     Ip6::Address destination;
 
-    (void)aMessageInfo;
-
-    VerifyOrExit(aHeader.GetCode() == Coap::Header::kCodePost, error = kThreadError_Drop);
+    VerifyOrExit(aHeader.GetType() == Coap::Header::kTypeConfirmable &&
+                 aHeader.GetCode() == Coap::Header::kCodePost, error = kThreadError_Drop);
 
     otLogInfoArp("Received address error notification\n");
 
+    if (!aMessageInfo.GetSockAddr().IsRealmLocalAllRoutersMulticast())
```

Comment:
I think this will work, but to be fully spec compliant, we should just check against `IsMulticast()`.  The Thread spec indicates:

> A Thread device MUST NOT send CoAP responses to multicast ADDR_ERR.ntf CoAP requests